src/db/db-helper.js

```diff
@@ -241,22 +245,92 @@ module.exports = class DBHelper {
 
   static async updateEventStatusWithdrawing(currBlockTime) {
     try {
-      await db.Events.update(
+      const updatedEvents = await db.Events.update(
         {
           txStatus: TX_STATUS.SUCCESS,
           $not: { status: EVENT_STATUS.WITHDRAWING },
           arbitrationEndTime: { $lte: currBlockTime },
           currentRound: { $gt: 0 },
         },
         { $set: { status: EVENT_STATUS.WITHDRAWING } },
-        { multi: true },
+        {
+          multi: true,
+          returnUpdatedDocs: true,
+        },
       );
+      await DBHelper.updateLeaderboardWithdrawing(updatedEvents);
     } catch (err) {
       logger.error(`UPDATE Event Status Withdrawing error: ${err.message}`);
       throw err;
     }
   }
 
+  static async updateLeaderboardWithdrawing(events) {
+    try {
+      each(events, async (event) => {
+        // get all the participants
+        const bets = await DBHelper.findBet({ eventAddress: event.address });
+        const filtered = [];
+        each(bets, (bet) => {
+          if (!find(filtered, {
+            eventAddress: bet.eventAddress,
+            betterAddress: bet.betterAddress,
+          })) {
+            filtered.push(bet);
+          }
+        });
+        // calculate winnings for all participants
+        const promises = [];
+        const winnings = [];
+        each(filtered, (bet) => {
+          promises.push(new Promise(async (resolve, reject) => {
+            try {
+              const { eventAddress, betterAddress } = bet;
+              const res = await MultipleResultsEventApi.calculateWinnings({
+                eventAddress,
+                address: betterAddress,
+              });
+              winnings.push({
+                eventAddress,
+                betterAddress,
+                amount: res.toString(10),
+              });
+              resolve();
+            } catch (err) {
+              reject(err);
+            }
+          }));
+        });
+        await Promise.all(promises);
+        // update leaderboard
+        each(winnings, async (winning) => {
+          // update event leaderboard winnings
+          const eventLeaderboardEntry = new EventLeaderboard({
+            eventAddress: winning.eventAddress,
+            userAddress: winning.betterAddress,
+            investments: 0,
```

Comment:
This is always 0. You should handle this when you fetch all the bets for this event on L274 by accumulating all the bets for each individual user and just stick it here.

Reply:
0 here is just to avoid duplicate addings. Event leaderboard entry should already have the investment for this user.